yarn-project/sequencer-client/src/sequencer/sequencer.ts

```diff
@@ -392,14 +393,13 @@ export class Sequencer {
     return true;
   }
 
-  setState(proposedState: SequencerState, force: boolean = false) {
+  setState(proposedState: SequencerState, secondsIntoSlot: number, force: boolean = false) {
     if (this.state === SequencerState.STOPPED && force !== true) {
       this.log.warn(
         `Cannot set sequencer from ${this.state} to ${proposedState} as it is stopped. Set force=true to override.`,
       );
       return;
     }
-    const secondsIntoSlot = getSecondsIntoSlot(this.l1GenesisTime, this.aztecSlotDuration);
```

Comment:
Why not pass in the current slot number, and keep the `getSecondsIntoSlot` call here? Seems like it would reduce some verbosity by not having to manually call `getSecondsIntoSlot` before each `setState`.